lib/npg_pipeline/cache/barcodes.pm

```diff
@@ -230,6 +232,25 @@ sub _get_tag_length {
   return $key;
 }
 
+# Add suffix to an index of the form "AAAAAA-GGGGGG"
+# to bring to a length of the form [4,6]
+sub _add_suffix {
+  my ($self, $index, $index_lengths) = @_;
+  my @idx = split /-/smx, $index;
+  my @ilengths = @{$index_lengths};
+  # if there is only one index length set the second index length to 0
+  if (scalar @ilengths == 1) { push @ilengths, 0; }
+  # if there is no separator in the index set the second part of the index to empty
+  if (scalar @idx == 1) { push @idx, q{}; }
+  # add suffix if required
+  if (length $idx[0] < $ilengths[0]) { $idx[0] .= $TAG_SUFFIX; }
+  if ( (length $idx[1] < $ilengths[1]) && length $idx[1]) { $idx[1] .= $TAG2_SUFFIX; }
```

Comment:
This will depend on whether the i5 is opposite or not, no? I.e. you'll need a `I5OPPOSITE_TAG2_SUFFIX` in that case...

Reply:
I think I've already replied to this elsewhere. The problem is we don't know the sequence to pad the i5 with if it is opposite. I suggest we write code to check the direction but if the i5 is opposite we simply output an error. As soon as we have lane which generates the error we can work out what sequence to pad the tags with.